### PR TITLE
es: Translations - Day 1 morning

### PR DESCRIPTION
**Pull request by henrif75**
Part of #284

### CODE REVIEW

File: po/es.po [hunk 1]
```diff
@@ -2647,55 +2642,56 @@ msgid ""
 "call functions such as `slice::get_unchecked` which does not do bounds "
 "checking."
 msgstr ""
-"El chequeo de límites no puede ser desactivado con el flag de compilador. "
-"Tampoco puede deshabilitarse directamente usando `unsafe`. Sin embargo, "
-"`unsafe` permite que llames a la función como `slice::get_unchecked` que no "
-"brinda chequeos de límites."
+"La comprobación de límites no se puede inhabilitar con una bandera de "
```
Comment (deavid) [inline po/es.po:2645]: At least to me 'inhabilitar' has a different meaning (https://www.fundeu.es/consulta/inhabilitar-y-deshabilitar-609/). 'deshabilitar' has widespread use, or 'desactivar' might be also a good replacement

File: po/es.po [hunk 2]
```diff
@@ -2647,55 +2642,56 @@ msgid ""
...
 
 #: src/why-rust/modern.md:8
 msgid "Generics."
-msgstr "Genericos."
+msgstr "Genéricos"
 
 #: src/why-rust/modern.md:9
 msgid "No overhead FFI."
-msgstr "No overhead FFI."
+msgstr "No hay sobrecarga de FFI"
```
Comment (deavid) [inline po/es.po:2670]: 'sobrecarga' is 'overload', not overhead. No overhead FFI means no added performance penalty when calling FFI.

File: po/es.po [hunk 3]
```diff
@@ -2647,55 +2642,56 @@ msgid ""
...
+msgstr "Abstracciones sin coste."
 
 #: src/why-rust/modern.md:12
 msgid "Tooling"
 msgstr "Herramientas"
 
 #: src/why-rust/modern.md:14
 msgid "Great compiler errors."
-msgstr "Gran compilador de errores."
+msgstr "Grandes errores del compilador."
```
Comment (deavid) [inline po/es.po:2682]: "Great compiler errors." here means that the compiler errors are really good and nice, not that they are big.

File: po/es.po [hunk 4]
```diff
@@ -2704,20 +2700,21 @@ msgid ""
 "writing a loop using `for` should result in roughly the same low level "
 "instructions as using the `.iter().fold()` construct."
 msgstr ""
-"Sin costo en abstracciones, similar a C++, quiere decir que no tienes que "
-"‘pagar’ por tener un un lenguaje de alto nivel con memoria o CPU. Por "
-"ejemplo, escribiendo un loop usando `for` debe resultar a bajo nivel en una "
-"instrucción que usa `.iter().fold()`."
+"Las abstracciones sin coste, similares a las de C++, significan que no "
+"tienes que 'pagar' por construcciones de programación de nivel superior con "
```
Comment (deavid) [inline po/es.po:2704]: programación de alto nivel

File: po/es.po [hunk 5]
```diff
@@ -2859,7 +2859,7 @@ msgstr ""
 
 #: src/basic-syntax/scalar-types.md:7
 msgid "Floating point numbers"
-msgstr ""
+msgstr "Números de punto flotante"
```
Comment (deavid) [inline po/es.po:2862]: "Números de coma flotante" - https://es.wikipedia.org/wiki/Coma_flotante

File: po/es.po [hunk 6]
```diff
@@ -2907,36 +2907,37 @@ msgstr ""
 
 #: src/basic-syntax/scalar-types.md:12
 msgid "The types have widths as follows:"
-msgstr "Los tipos tienen una capacidad así:"
+msgstr "Los tipos tienen la siguiente anchura:"
 
 #: src/basic-syntax/scalar-types.md:14
 msgid "`iN`, `uN`, and `fN` are _N_ bits wide,"
-msgstr "`iN`, `uN`, and `fN` son _N_ bits de capacidad,"
+msgstr ""
```
Comment (deavid) [inline po/es.po:2914]: can you provide a translation fix instead of removing it?

File: po/es.po [hunk 7]
```diff
@@ -3057,40 +3054,39 @@ msgid ""
 "the debug output. We could also have used `{a}` and `{a:?}` without "
 "specifying the value after the format string."
 msgstr ""
-"En la función main, la palabra print pregunta por la implementación de debug "
-"con `?` para el formato de parámetro: `{}` que da una salida estándar, `{:?}"
-"`. También podemos usar `{a}` y \\`{a:?} sin especificar los valores luego "
-"del formato string."
+"En la función principal, la instrucción de impresión solicita la "
```
Comment (deavid) [inline po/es.po:3057]: "main" is the proper noun and name of the main() function, better leave it untranslated.

File: po/es.po [hunk 8]
```diff
@@ -3130,29 +3127,31 @@ msgstr ""
...
+"invocar métodos (prueba `ref_x.count_ones()`)."
 
 #: src/basic-syntax/references.md:19
 msgid ""
 "References that are declared as `mut` can be bound to different values over "
 "their lifetime."
-msgstr "Las referencias se declaran como `mut` y pueden cambiar de valor."
+msgstr ""
+"Las referencias que se declaran como `mut` se pueden vincular a distintos "
+"valores durante su  tiempo de vida."
```
Comment (deavid) [inline po/es.po:3154]: double space in " su  "

**Review by deavid — CHANGES_REQUESTED** [review]

**Comment by deavid** [comment]
Overall I see this PR as a huge improvement to the Spanish translation. However I'm not very keen on translating every programming specific term into English, and we should be a bit more careful on making this suitable too for Spanish speakers outside of Spain.

File: po/es.po [hunk 9]
```diff
@@ -2221,146 +2223,145 @@ msgid ""
...
 
 #: src/welcome-day-1/what-is-rust.md:16
 msgid "servers."
-msgstr "servers."
+msgstr "servidores."
 
 #: src/welcome-day-1/what-is-rust.md:21
 msgid "Rust fits in the same area as C++:"
-msgstr "Rust encaja en el mismo área de C++:"
+msgstr "Rust se asemeja bastante a C++:"
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/es.po:2263]: Thanks, all this translation came from the localization services team. Can you suggest a better translation (I'm not fluent in Spanish :) )

File: po/es.po [hunk 10]
```diff
@@ -2221,146 +2223,145 @@ msgid ""
...
 "Can be scaled down to very constrained devices such as microcontrollers."
-msgstr "Puede usarse en dispositivos limitados como teléfonos celulares"
+msgstr ""
+"Se puede reducir verticalmente a dispositivos muy limitados, como los "
+"microcontroladores."
 
 #: src/welcome-day-1/what-is-rust.md:26
 msgid "Has no runtime or garbage collection."
-msgstr "No tiene runtime o garbage colector."
+msgstr "No tiene tiempos de ejecución ni recolección de memoria residual."
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/es.po:2282]: Do you suggest keeping these terms in English italicized?

[the remark this answers is not in the document]
Comment (deavid) [reply po/es.po:2282]: yes, italicized is the recommendation from RAE for dealing with foreign words. I recommend doing that instead.

File: po/es.po (hunk 9 above)
[the remark this answers is not in the document]
Comment (deavid) [reply po/es.po:2263]: "Rust satisface las mismas necesidades que C++:" might be a better wording although it sounds different than english, it seems to convey the same thing.

File: po/es.po [hunk 11]
```diff
@@ -2400,103 +2410,88 @@ msgid ""
...
 "Explain that all variables are statically typed. Try removing `i32` to "
 "trigger type inference. Try with `i8` instead and trigger a runtime integer "
 "overflow."
 msgstr ""
-"Explica que todas las variables son de tipo estáticas. Prueba removiendo "
-"`i32` para que se dispare la inferencia de tipos. Intenta con `i8` y ejecuta "
-"un overflow de integer."
+"Explica que todas las variables están tipadas estáticamente. Prueba a quitar "
+"`i32` para activar la inferencia de tipos. Prueba con `i8` y activa un "
+"desbordamiento de enteros con tiempo de ejecución."
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/es.po:2432]: I also restored the term "runtime".

File: po/es.po [hunk 12]
```diff
@@ -2634,11 +2627,13 @@ msgid ""
...
-"El overflow en Integer es definido vía un flan en tiempo de compilación. Las "
-"opciones son alertas panic (un crash controlado del programa) o una "
-"semántica que lo envuelve. Por defecto, obtienes el alerta en modo "
-"depuración (`cargo build`) y envolviéndolo en el modo release (`cargo build —"
-"release`)."
+"El desbordamiento de enteros se define mediante la bandera de tiempo\n"
+"de compilación [`overflow-checks`](https://doc.rust-lang.org/rustc/codegen-"
+"options/index.html#overflow-checks). Si se habilita, el programa entrará en "
+"pánico (un fallo controlado del programa). De lo contrario, se obtendrá una "
+"semántica envolvente. De forma predeterminada, se obtienen pánicos en el "
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/es.po:2634]: I restored the wrap-around term.

**Comment by henrif75** [comment]
Thanks so much @deavid for your thorough review. I've made all your suggested changes, PTAL.

**Comment by deavid** [comment]
I reviewed two more in the meantime. I'll be OOO for the next 2 weeks. I'll try to push some reviews before going OOO, but on next monday you might need to reach out to a different reviewer to get unblocked.

**Review by deavid — APPROVED** [review]

**Comment by henrif75** [comment]
> I reviewed two more in the meantime. I'll be OOO for the next 2 weeks. I'll try to push some reviews before going OOO, but on next monday you might need to reach out to a different reviewer to get unblocked.

Thanks again!

**Pull request merged**